connections/tests/utils.py

```diff
@@ -12,29 +13,94 @@ def make_apartments_sold_in_elastic() -> None:
     sleep(3)
 
 
-def get_elastic_apartments_for_sale_uuids() -> list:
+def get_elastic_apartments_for_sale_uuids(
+    etuovi=False, oikotie=False, only=False
+) -> list:
     s_obj = (
         Search()
         .query("match", _language="fi")
         .query("match", apartment_state_of_sale=ApartmentStateOfSale.FOR_SALE)
     )
+
+    if oikotie:
+        s_obj = s_obj.query("match", publish_on_oikotie=oikotie)
+        if only:
+            s_obj = s_obj.query("match", publish_on_etuovi=False)
+    if etuovi:
+        s_obj = s_obj.query("match", publish_on_etuovi=etuovi)
+        if only:
+            s_obj = s_obj.query("match", publish_on_oikotie=False)
+
+    if only and not etuovi and not oikotie:
+        s_obj = s_obj.query("match", publish_on_oikotie=oikotie).query(
+            "match", publish_on_etuovi=etuovi
+        )
+
+    if etuovi and oikotie and only:
+        # get apartments only by published flag
+        s_obj = (
+            Search()
+            .query("match", publish_on_oikotie=oikotie)
+            .query("match", publish_on_etuovi=etuovi)
+            .query("match", apartment_state_of_sale=ApartmentStateOfSale.RESERVED)
+        )
```

Comment:
This is way too complex if clause. This part or the whole function needs to chop into smaller functions. We can take a call for this.